3 - Assignment/240714 - Rust Basic + Linux Kernel, Week 2, Day 2/nested-arrays/src/lib.rs

```diff
@@ -1,5 +1,9 @@
 pub fn transpose(matrix: [[i32; 3]; 3]) -> [[i32; 3]; 3] {
-    todo!("Implement this")
+    [
+        [matrix[0][0], matrix[1][0], matrix[2][0]],
+        [matrix[0][1], matrix[1][1], matrix[2][1]],
+        [matrix[0][2], matrix[1][2], matrix[2][2]],
+    ]
```

Comment:
틀린 건 아니지만, 이후에 배열의 크기가 변했을 때는 코드를 다시 작성해야 하는 문제가 있습니다.
유지 보수를 위해서 다음과 같이 작성하는 게 좀 더 좋겠네요.
```suggestion
    let mut ret = [[0; 3]; 3];

    for i in 0..3 {
        for j in 0..3 {
            ret[i][j] = matrix[j][i];
        }
    }

    ret
```